lib/SGN/Controller/AJAX/QualityControl.pm

```diff
@@ -45,28 +45,57 @@ sub prepare: Path('/ajax/qualitycontrol/prepare') Args(0) {
     my $schema = $c->dbic_schema("Bio::Chado::Schema", "sgn_chado");
     my $temppath = $c->config->{basepath}."/".$tempfile;
 
-    my $ds = CXGN::Dataset::File->new(people_schema => $people_schema, schema => $schema, sp_dataset_id => $dataset_id, exclude_dataset_outliers => 1, file_name => $temppath, quotes => 0);
-    $ds->retrieve_phenotypes();
-    my $pf = CXGN::Phenotypes::File->new( { file => $temppath."_phenotype.txt" });
+    my $dbh = $c->dbc->dbh();
 
-    # my @traits_select = ();
-    my $traits = $pf->traits();
+    my $trait_sql = "SELECT dataset FROM sgn_people.sp_dataset WHERE sp_dataset_id = $dataset_id";
```

Comment:
Use CXGN::Dataset->new( ... ... dataset_id => ... ) and then use accessor to retrieve the jason